src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -217,7 +217,8 @@ static AppInstallationToken generateAppInstallationToken(
       } else {
         appInstallation =
             appInstallations.stream()
-                .filter(installation -> installation.getAccount().getLogin().equals(owner))
+                .filter(
+                    installation -> installation.getAccount().getLogin().equalsIgnoreCase(owner))
```

Comment:
what locales is would GitHub (enterprise) be running in?  is it always English, or could someone set this to turkish, in which case would this incorrectly match some items.

for example `String.equalsIgnoreCase` will return true for `ivan` and `Ivan`  but if the GitHub user was `İvan` then they could be a different user to `Ivan` but equalsIgnoreCase will say they are all the same.

Ivan = ivan -> true
İvan = ivan -> true
İvan = Ivan -> true
Ivan = ıvan -> true

Reply:
As far as I can tell there is no way to set a GitHub (Enterprise) instance to another locale, but there is no documentation stating this as a fact. When creating an organization, you can enter arbitrary unicode for the display name, but it is normalized to alphanumeric `a-Z0-9` in the ID:

![Screenshot from 2022-03-07 16-33-08](https://user-images.githubusercontent.com/16737117/157121962-58b4fd05-477b-4d79-b12f-ce018a8b74aa.png)

Signup on github.com is clear on its requirements:

![image](https://user-images.githubusercontent.com/16737117/157122431-d2e1f3fe-f1a6-46a1-98dd-7f3a8f8683a0.png)


Unfortunately GitHub doesn't expose an API to determine if a given string is a valid username and provides no docs on the matter. The fact that user and organization IDs are case insensitive doesn't seem to be documented anywhere by GitHub, so we probably want to take a pessimistic approach if we do anything here.

Reply:
Very unlikely but possible. You can replace `x.equalsIgnoreCase(y)` with `x.toLowerCase(Locale.ROOT).equals(y.toLowerCase(Locale.ROOT))`.

Reply:
FYA: https://bugs.openjdk.java.net/browse/JDK-8283000